### DISCUSSION TITLE
New secondary source type: reverse geocoding

### DISCUSSION THREAD

**Discussion by Arcticn** [breezy-weather/breezy-weather#1258] [Ideas]
This app is incredibly great, really loves it!

After compare weather sources, I choose Open-Meteo as my main source. However, this source will only show "Current location", while AccuWeather can show detailed location. I read the doc and learned Open-Meteo didn't support Reverse geocoding now, which causes this problem.

And my problem is, I want to use "China" as my precipitation nowcasting provider. This will work good when using AccuWeather, because the app got accurate location, learning I'm in China, and unlocks the "China" provider. However, using Open-Meteo will not unlock "China", because the app doesn't know where you are.

Is my guess right? If it's true, can you add an option to let AccuWeather get the location while using Open-Meteo as main source so that some specific regional provider will be unlocked, like "China"? 

Morever, you can use this location to replace "Current location", just like when you're using Accuweather.

Thanks!

### Acknowledgements

- [x] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.7 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
We do have an offline reverse geocoder that is able to guess your country. You should be able to select the China source when your location is in Mainland China. Let me know if that's not the case.

**Reply by Arcticn** [reply]
Yes, I think it's not working, you can check this video. When using Accuweather, "china" is available, however, this option disappeared when using Open-Meteo

https://github.com/user-attachments/assets/dd4e4180-6e1b-46d8-aa78-aad8d4757505

**Reply by Arcticn** [reply]
Further information: I found that when using IP.SB instead of GPS as location service, "China" option comes back, but the location is far from me(of course still in China), so I won't use it.

**Reply by papjul** [reply]
I don't reproduce your issue when "fake travelling" in the middle of Mainland China. Are you living in a territorialy disputed place?

**Reply by papjul** [reply]
> Vey close to the sea shore

1. Ah, is it possible Android locates you in the sea (so outside the frontiers of the country)? Can you check in a different app where it locates you?
2. Do you have the same issue when you retry more in town?

**Reply by Arcticn** [reply]
I tried navigation app and the default weather app, location is correct.

Also, I tried to chang the main source to other worldwide sources with Reverse geocoding: AccuWeather | Météo-France | DMI | Meteo AM. Three out of four sources returns correct loacation and "China" is unlocked, only Meteo AM "failed to parse weather data". Météo-France gives the most detailed position among these sources, and is correct.

**Reply by papjul** [reply]
What about the 2nd question?

**Reply by Arcticn** [reply]
So I tried "fake travelling" in android emulator, I found that in my position, you need to live about 1.5km away from the sea shore to unlock "China" option, also exists in other places in China, but the distance differs.

**Reply by papjul** [reply]
Thank you for your investigation! It's curious because our source for offline reverse geocoding should have a 50 m precision: https://www.naturalearthdata.com/downloads/50m-cultural-vectors/

#955 will probably help in the future. In the meantime I believe that even if it is shown as `china (unavailable)`, it should still work anyway, right?

**Reply by Arcticn** [reply]
Thank you! That's truly strange. And yes, it still works anyway.

So, is it a good idea to add an option, which choose an extra secondary weather source to help with reverse geocoding?  This can also help replace "Current location" with detailed location.

**Reply by papjul** [reply]
I need to think about it. I was more thinking of a global setting to select which reverse geocoding source you want to use for all locations without this feature. In the current state, that's probably the same as what you suggest since you only need reverse geocoding for one location: current location.

But locations pinpointed manually from a map will be affected once we implement #10

**Reply by Arcticn** [reply]
Okay, thank you for work, have a nice day!

**Comment by papjul** [comment]
Implemented in 29ed15fad7b7e6f6bb6d7bb1d041ba406bf375c1

It will be available in v5.4.3, on Current location, in Edit > Sources > Address lookup.
